src/jit/morph.cpp

```diff
@@ -12767,69 +12767,80 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
         case GT_GE:
         case GT_GT:
 
-            if ((tree->gtFlags & GTF_UNSIGNED) == 0)
+            if (op2->gtOper == GT_CNS_INT)
             {
-                if (op2->gtOper == GT_CNS_INT)
+                bool isOpSigned = (tree->gtFlags & GTF_UNSIGNED) == 0;
```

Comment:
use `tree->IsUnsigned()`

Reply:
And remove the `isOpSigned` variable. There's no point in doing CSE for the native C++ compiler...